File: framework/src/play/src/main/java/play/Mode.java
/*
 * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
 */
package play;

/**
 * Application mode, either `DEV`, `TEST`, or `PROD`.
 */
public enum Mode {
    DEV, TEST, PROD;

    public play.api.Mode.Mode asScala() {
        if (this == DEV) {
            return play.api.Mode.dev();
        } else if (this == PROD) {
            return play.api.Mode.prod();
        }
        return play.api.Mode.test();
    }
}

File: core/play/src/main/scala/play/api/Mode.scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package play.api

/**
 * Application mode, either `Dev`, `Test`, or `Prod`.
 *
 * @see [[play.Mode]]
 */
sealed abstract trait Mode

object Mode {
  case object Dev  extends Mode
  case object Test extends Mode
  case object Prod extends Mode

  lazy val values: Set[play.api.Mode] = Set(Dev, Test, Prod)
}
